beta/src/content/apis/react/memo.md

```diff
@@ -2,21 +2,314 @@
 title: React.memo
 ---
 
-<Wip>
+<Intro>
 
-This section is incomplete, please see the old docs for [React.memo.](https://reactjs.org/docs/react-api.html#reactmemo)
+`memo` lets you avoid re-rendering a component when its props are unchanged.
 
-</Wip>
+```
+export default memo(function Button({title}) {
+  return <button>{title}</button>
+});
+```
 
+</Intro>
 
-<Intro>
+- [Usage](#usage)
+  - [Avoiding re-rendering when props are unchanged](#avoiding-rerendering-when-props-are-unchanged)
+  - [Updating a memoized component using state](#updating-a-memoized-component-using-state)
+  - [Updating a memoized component using a context](#updating-a-memoized-component-using-state)
+  - [Minimizing props changes](#minimizng-props-changes)
+  - [Specifying a custom comparison function](#specifying-a-custom-comparison-function)
+- [Reference](#reference)
+- [Troubleshooting](#troubleshooting)
+  - [My component re-renders when a prop is an object or array](#my-component-rerenders-when-a-prop-is-an-object-or-array)
+  - [My component re-renders when a prop is a function](#my-component-rerenders-when-a-prop-is-a-function)
+
+---
+
+## Usage {/*usage*/}
+
+### Avoiding re-rendering when props are unchanged {/*avoiding-rerendering-when-props-are-unchanged*/}
+
+React normally re-renders a component whenever its parent re-renders. With `memo`, you can create a component that React will not re-render when its parent re-renders so long as its new props are the same as the old props. Such a component is said to be *memoized*.
+
+`memo` is a function: you use it by wrapping your component in a call to `memo` and using the value that it returns in place of your original component:
+
+```
+export default memo(function Greeting({name}) {
```

Comment:
there are several common styles

```js
function Foo() { }
export default memo(Foo)
```


```js
let Foo = memo(() => { })

export default Foo
```


```js
function Foo() { }
Foo = memo(Foo)
export default Foo
```

and there also different things you can do with named ones.

should we stick with some single style consistently? which one and why?

Reply:
ideally the syntax we'd use would work consistently whether you render a memo'd component in the same file or export it

Reply:
The big advantage of your second example is that the memoization is co-located with the component definition and not snuck in someplace else where it's hard to see.

Does having it be an anonymous function mess up debugging?

Reply:
Yeah it's not ideal for debugging. I guess

```js
let Foo = memo(function Foo() {

})
```

would also work even if a bit clunky

Reply:
Could we have a transform that assigns a name to a function if it's passed to memo based on what the memo gets assigned to?

Reply:
In theory yes, but in practice we haven't made/encouraged one